kernprof.py

```diff
@@ -155,53 +155,46 @@ def find_script(script_name):
     raise SystemExit(1)
 
 
-def main(args=None):
-    def strictly_positive(option, opt, value, parser):
-        if value <= 0:
-            raise optparse.OptionValueError("option %s: floating-point value must be > 0, got %s" % (opt, value))
-        setattr(parser.values, option.dest, value)
-
-    if args is None:
-        args = sys.argv
-    usage = "%prog [-s setupfile] [-o output_file_path] scriptfile [arg] ..."
-    parser = optparse.OptionParser(usage=usage, version=__version__)
-    parser.allow_interspersed_args = False
-    parser.add_option('-l', '--line-by-line', action='store_true',
-        help="Use the line-by-line profiler from the line_profiler module "
-        "instead of Profile. Implies --builtin.")
-    parser.add_option('-b', '--builtin', action='store_true',
-        help="Put 'profile' in the builtins. Use 'profile.enable()' and "
-            "'profile.disable()' in your code to turn it on and off, or "
-            "'@profile' to decorate a single function, or 'with profile:' "
-            "to profile a single section of code.")
-    parser.add_option('-o', '--outfile', default=None,
-        help="Save stats to <outfile>")
-    parser.add_option('-s', '--setup', default=None,
+def main():
+    def positive_float(value):
+        val = float(value)
+        if val <= 0:
+            raise ArgumentError
+        return val
+
+    parser = ArgumentParser()
```

Comment:
Can you populate the `description` parameter for `ArgumentParser`? It probably makes sense to leave `prog` and `usage` as the defaults right? 

I'm also not sure what `parser.allow_interspersed_args` does, can you verify that its False behavior is the same as argparse's?

Reply:
prog and usage are generated by argparse so they are complete, and we don't have to maintain a usage string in sync with the actual arguments.

Reply:
That's good enough for me on usage and prog.

It looks like to emulate `allow_interspersed_args=False` we should be using `parser.parse_intermixed_args` or `parse_known_intermixed_args`: https://bugs.python.org/issue15258 We could optionally just ignore this and use basic argparse. I'd be fine with doing that. It may be simpler and have low-risk of breaking code. 

I think a description would be useful though. Just something like `description='profile your code with kernprof'` would be sufficient.

Reply:
I added a basic description.
For the interspersed args, the REMAINDER argument (`nargs='...'`) handles this and allow to pass the additional arguments to the script (all arguments after the script path are for the script and it works perfectly).